python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+        if len(image_array.shape) == 2:
+            # Single donut: [cropSize, cropSize] -> [1, cropSize, cropSize]
+            image_array = image_array.reshape(1, self.cropSize, self.cropSize)
+        elif len(image_array.shape) == 3:
+            # Multiple donuts: [num_stamps, cropSize, cropSize] - correct shape
+            pass
+        else:
+            raise ValueError(f"Unexpected image array shape: {image_array.shape}")
+
+        num_stamps = image_array.shape[0]
+        self.log.info("Processing %d donut stamp(s) of size %dx%d pixels",
+                     num_stamps, self.cropSize, self.cropSize)
+        self.log.debug("DONUT PROCESSING DEBUG: Starting donut stamp creation")
+        self.log.debug("Normalized to %d stamp(s) of size %dx%d", num_stamps, self.cropSize, self.cropSize)
+        # Get centroid positions using helper method
+        cent_x_list, cent_y_list = self._calculate_centroid_positions(exposure, num_stamps)
+
+        # Create metadata for all donuts
+        metadata = PropertyList()
+        metadata["RA_DEG"] = [0.0] * num_stamps  # Will be calculated from WCS
+        metadata["DEC_DEG"] = [0.0] * num_stamps  # Will be calculated from WCS
+        metadata["CENT_X"] = cent_x_list  # Use TARTS centers or exposure center
+        metadata["CENT_Y"] = cent_y_list  # Use TARTS centers or exposure center
+        metadata["DET_NAME"] = [detectorName] * num_stamps  # Valid detector name
+        metadata["CAM_NAME"] = [cameraName] * num_stamps  # Valid camera name
+        metadata["DFC_TYPE"] = [defocalType] * num_stamps  # "extra" or "intra"
+        metadata["DFC_DIST"] = [1.5] * num_stamps  # Default defocal distance in mm
+        metadata["BANDPASS"] = [bandLabel] * num_stamps  # Filter band
+        metadata["BLEND_CX"] = ["nan"] * num_stamps  # No blended sources
+        metadata["BLEND_CY"] = ["nan"] * num_stamps  # No blended sources
+
+        # Add TARTS-specific metadata using helper method
+        try:
+            fx_list = self._extract_and_normalize_tarts_data('fx', num_stamps)
+            fy_list = self._extract_and_normalize_tarts_data('fy', num_stamps)
+            metadata["FX"] = fx_list
+            metadata["FY"] = fy_list
+        except AttributeError:
+            self.log.warning("TARTS fx/fy attributes not available, using default values")
+            metadata["FX"] = [0.0] * num_stamps  # Default fx values
+            metadata["FY"] = [0.0] * num_stamps  # Default fy values
+
+        # Create list of DonutStamp objects
+        donutStamps = []
+
+        for i in range(num_stamps):
+            # Create MaskedImageF for this donut
+            stamp_im = afwImage.MaskedImageF(self.cropSize, self.cropSize)
+            stamp_im.image.array[:] = image_array[i]  # Use the i-th donut image
+            stamp_im.setXY0(0, 0)  # Set origin
+
+            # Create linear WCS for this donut stamp using TARTS center if
+            # available
+            try:
```

Comment:
This whole sequence seems like it could be simplified and cleaned up a bit. You have three else statements that apply the same center, you may consider only handling the case differently when the conditions are met and otherwise default to the centroid_position